### ISSUE TITLE
using tailf with argcomplete breaks console

### ISSUE THREAD

**Issue by MatsNilsson70** [kislyuk/argcomplete#146]
After "pip install argcomplete" and "activate-global-python-argcomplete" 
tailf breaks console.

To reproduce:
Use tab when entering below command:
tailf /var/log/messages 
and then "Ctrl-c" to jump out of tailf
After that the console is destroyed, not seeing what you are writing.....

### System info(also reproducable on SUSE 12):
uname -r
3.13.0-96-generic

cat /etc/os-release
NAME="Ubuntu"
VERSION="14.04.5 LTS, Trusty Tahr"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 14.04.5 LTS"
VERSION_ID="14.04"
HOME_URL="http://www.ubuntu.com/"
SUPPORT_URL="http://help.ubuntu.com/"
BUG_REPORT_URL="http://bugs.launchpad.net/ubuntu/"

pip2 --version
pip 1.5.4 from /usr/lib/python2.7/dist-packages (python 2.7)

pip2 list | grep argcomplete
argcomplete (1.4.1)

**Comment by kislyuk** [comment]
I've seen something similar happen to my terminal - echo is occasionally completely broken in the terminal after quitting programs; you have to type `reset<Enter>` to restore the terminal - but I'm not certain if it's argcomplete's fault or why it happens. If it is argcomplete's fault, it may have something to do with how `complete -D` works or some other details of https://github.com/kislyuk/argcomplete/blob/master/argcomplete/bash_completion.d/python-argcomplete.sh. Does anyone have any ideas?

**Comment by MatsNilsson70** [comment]
Thanks for the answer, our users is using tailf very much to see logs of the system.
So I can not use argcomplete if I do not get a solution of this.
I have tested by rewriting the python-argcomplete.sh with no success.
I tested with pointing out the file in python-argcomplete.sh :
complete -o nospace -o default -o bashdefault  -F _python_argcomplete_global our_python_script.py
In that case both tailf and our script works but if you do it like this it still fails:
>our_python_script.py (and using tab, but never press enter)
erase the command and write 
>tailf /var/log/messages (using tab)
It seems that it is executing python-argcomplete.sh because you never pressed enter

Something in this argcomplete implementation is messing up tailf.
Please help me out, anyone?

**Comment by MatsNilsson70** [comment]
Some more info:
It seems that if you execute any bash commands in python-argcomplete.sh it will fail the tailf command.
E.g. head or grep or simular,
Why I do not understand?
For now I use this "hardcoded" python-argcomplete.sh  and it seems to work:


```
# Copyright 2012-2013, Andrey Kislyuk and argcomplete contributors.
# Licensed under the Apache License. See https://github.com/kislyuk/argcomplete for more info.

# Copy of __expand_tilde_by_ref from bash-completion
__python_argcomplete_expand_tilde_by_ref () {
    if [ "${!1:0:1}" = "~" ]; then
        if [ "${!1}" != "${!1//\/}" ]; then
            eval $1="${!1/%\/*}"/'${!1#*/}';
        else
            eval $1="${!1}";
        fi;
    fi
}
_python_argcomplete_global() {
    local executable=$1
    __python_argcomplete_expand_tilde_by_ref executable

    local IFS=$(echo -e '\v')
    COMPREPLY=( $(_ARGCOMPLETE_IFS="$IFS" \
              COMP_LINE="$COMP_LINE" \
              COMP_POINT="$COMP_POINT" \
              _ARGCOMPLETE_COMP_WORDBREAKS="$COMP_WORDBREAKS" \
              _ARGCOMPLETE=$ARGCOMPLETE \
              "$executable" 8>&1 9>&2 1>/dev/null 2>&1) )
    if [[ $? != 0 ]]; then
       unset COMPREPLY
    fi
}
complete -o nospace -o default -o bashdefault  -F _python_argcomplete_global example1.py
complete -o nospace -o default -o bashdefault  -F _python_argcomplete_global example2.py
```

The bad thing is that I can not dynamically load python files, e.g. by specifying PYTHON_ARGCOMPLETE_OK. and maybe other things...

**Comment by evanunderscore** [comment]
This is a bash bug that was fixed in 4.4:

> d.  Fixed a bug that could potentially cause the terminal attributes to be set incorrectly by a command run from a programmable completion.

Source: https://tiswww.case.edu/php/chet/bash/CHANGES
See also: http://stackoverflow.com/questions/39071003/bash-completion-sometimes-meshes-up-my-terminal-when-the-completion-function-rea

**Comment by kislyuk** [comment]
Thanks for the find, @evanunderscore. My evidence is consistent with this.
